Neos.Flow/Classes/Security/Policy/Roles.php

```diff
@@ -0,0 +1,116 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Security\Policy;
+
+use Neos\Flow\Annotations as Flow;
+
+/*
+* This file is part of the Neos.Flow package.
+*
+* (c) Contributors of the Neos Project - www.neos.io
+*
+* This package is Open Source Software. For the full copyright and license
+* information, please view the LICENSE file which was distributed with this
+* source code.
+*/
+
+/**
+ * A collection class containing all roles for an account
+ *
+ * @Flow\Proxy(false)
+ */
+final class Roles implements \JsonSerializable, \IteratorAggregate, \Countable, \ArrayAccess
+{
+    /**
+     * @var Role[]
+     */
+    private $roles;
+
+    private function __construct(array $roles)
+    {
+        $this->roles = $roles;
+    }
+
+    public static function create(): self
+    {
+        return new static([]);
+    }
+
+    public static function fromArray(array $roles): self
+    {
+        $processedRoles = [];
+        array_walk($roles, function ($role) use (&$processedRoles) {
+            if (!$role instanceof Role) {
+                throw new \InvalidArgumentException('Expected instance of Role. Was given ' . gettype($role));
```

Comment:
Missing exception code
```suggestion
                throw new \InvalidArgumentException(sprintf('Expected instance of Role. Given: %s', is_object($role) ? get_class($role) : gettype($role)), 1568888776);
```